src/libratbag-util.h

```diff
@@ -82,22 +82,24 @@ int list_empty(const struct list *list);
 
 static inline void
 cleanup_free(void *p) {
-	free(*(void**)p);
+	if (p)
+		free(*(void**)p);
 }
 
 static inline void
 cleanup_close(int *p) {
-	close(*p);
+	if (p)
+		close(*p);
 }
 static inline void
 cleanup_udev_unref(struct udev **udev) {
-	if (*udev)
+	if (udev && *udev)
 		udev_unref(*udev);
 }
 
 static inline void
 cleanup_udev_device_unref(struct udev_device **udev_device) {
-	if (*udev_device)
+	if (udev_device && *udev_device)
 		udev_device_unref(*udev_device);
 }
 
```

Comment:
tbh, for the cleanup functions I'd rather have an assert() than a null-check, there is no code path where we can pass null here